frontend/util.lua

```diff
@@ -400,4 +400,102 @@ function util.splitToArray(str, splitter, capture_empty_entity)
     return result
 end
 
+--- Convert a Unicode codepoint (number) to UTF8 char
+--
+--- @int c Unicode codepoint
+--- @treturn string UTF8 char
+function util.unicodeCodepointToUtf8(c)
+    if c < 128 then
+        return string.char(c)
+    elseif c < 2048 then
+        return string.char(192 + c/64, 128 + c%64)
+    elseif c < 55296 or 57343 < c and c < 65536 then
+        return string.char(224 + c/4096, 128 + c/64%64, 128 + c%64)
+    elseif c < 1114112 then
+        return string.char(240 + c/262144, 128 + c/4096%64, 128 + c/64%64, 128 + c%64)
+    else
+        return util.unicodeCodepointToUtf8(65533) -- U+FFFD REPLACEMENT CHARACTER
+    end
+end
+
+local HTML_ENTITIES_TO_UTF8 = {
+    ["&lt;"] = "<",
+    ["&gt;"] = ">",
+    ["&amp;"] = "&",
+    ["&quot;"] = '"',
+    ["&apos;"] = "'",
+    ["&nbsp;"] = "\xC2\xA0",
+    ["&#(%d+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x) ) end,
+    ["&#x(%x+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x,16) ) end,
```

Comment:
A more comprehensive list can be found at https://stackoverflow.com/questions/14899734/unescape-numeric-xml-entities-with-lua.
p.s. You may need to put &amp; at the end of the list, otherwise '&amp;quot;' will be converted to '"', but indeed it should be '&quot;'.

Why do not name this function "htmlDecode"?

According to the following usage, I would suggest to return a tuple from this function: `{string decoded, bool changed}`. i.e.
```
function util.htmlDecode(text)
  if text contains html entities then
    return decode(text), true
  else
    return text, false
  end
end function
```
then a `util.htmlDecodeIfEncoded(text)` can be written as,
```
function util.htmlDecodeIfEncoded(text)
  if text contains html tags then
    return text, false
  else
    local decoded, result = util.htmlDecode(text)
    if result then
      return decoded, true
    else
      return text, false
    end
  end
end function
```
So you won't need to handle the double encoded html in `util.htmlToPlainTextIfHtml()`.